src/models/room.ts

```diff
@@ -933,6 +1051,10 @@ export class Room extends TypedEventEmitter<EmittedEvents, RoomEventHandlerMap>
             .getState(EventTimeline.BACKWARDS);
         this.currentState = this.getLiveTimeline()
             .getState(EventTimeline.FORWARDS);
+
+        // Let people know to register listeners for the new state references
+        this.emit(RoomEvent.OldStateUpdated, this, previousOldState, this.currentState);
+        this.emit(RoomEvent.CurrentStateUpdated, this, previousCurrentState, this.currentState);
```

Comment:
Calling out this new behavior. See how it's used in `sync.ts` to re-register the state listeners.